editor/filesystem_dock.cpp

```diff
@@ -1839,7 +1840,7 @@ Variant FileSystemDock::get_drag_data_fw(const Point2 &p_point, Control *p_from)
 			all_not_favorites &= !is_favorite;
 			selected = tree->get_next_selected(selected);
 		}
-		if (all_favorites) {
+		if (!all_not_favorites) {
 			paths = _tree_get_selected(false);
 		} else {
 			paths = _tree_get_selected();
```

Comment:
@groud
Why is there this distinction? Or why is false not passed for non-favorites?
This makes the actual selection not contain the individual files if the dir containing them is also selected. I think this behavior is unintuitive especially since the visible selection can contain both the dir and the contained files. If the user wants only the dir they can select just that.

Reply:
No, I think the use case when you want both the containing folder and the children recorded as favorites is a pretty non-existing situation. I think it is is better to avoid confusing situation where a user unintentionally selected a bunch of files and end up with nested favorites for no real reasons.

Also, the variable name should be kept as before, negations in variable naming does not make a lot of sense.

Reply:
I just think that this
![sel_a](https://user-images.githubusercontent.com/25692790/67164661-4c6a3080-f37d-11e9-841b-00bb5ffdf42c.PNG)
shouldn't be the same as this
![sel_o](https://user-images.githubusercontent.com/25692790/67164680-889d9100-f37d-11e9-9022-48ece1c76107.PNG)

Reply:
Well, for making favorites, I think it should be the same. :)